integration_tests/camera_settings.cpp

```diff
@@ -0,0 +1,97 @@
+#include <iostream>
+#include <functional>
+#include <atomic>
+#include <future>
+#include "dronecore.h"
+#include "integration_test_helper.h"
+#include "camera_test_helpers.h"
+
+using namespace dronecore;
+using namespace std::placeholders; // for `_1`
+
+TEST(CameraTest, ShowSettings)
+{
+    DroneCore dc;
+
+    ConnectionResult ret = dc.add_udp_connection();
+    ASSERT_EQ(ret, ConnectionResult::SUCCESS);
+
+    // Wait for device to connect via heartbeat.
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+
+    Device &device = dc.device();
+    auto camera = std::make_shared<Camera>(device);
+
+    // Wait for download to happen.
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+
+    // Set to photo mode
+    set_mode(camera, Camera::Mode::PHOTO);
+
+    std::map<std::string, std::string> settings;
+    EXPECT_TRUE(camera->get_possible_settings(settings));
+
+    LogDebug() << "Possible settings in photo mode: ";
+    for (auto setting : settings) {
+        LogDebug() << "- " << setting.second;
```

Comment:
It looks if both name and value are printed. Isn't it ?

Reply:
I think so, I need to work on this, it's far from finished.